### PR TITLE
Add/Enable repositories

### PR DESCRIPTION
**Pull request by johnsgill3**
Wrote helper routine to simply create new repository and set ensure necessary parameters are setup.

Wrote routine for repository model to enable a GitHub requests to get all the files.

TODO Items:
- [ ] Database Error Handling.
- [ ] Should master branch be one always queried for initial files?
- [ ] Handle GitHub not found error if master branch doesn't exist.
- [x] Write tests for new routines once GitHub(#35) and OmniAuth(#42) mock env are available.
- [ ] Handle disable repository. This might have to be addressed by #32

### CODE REVIEW

File: app/models/repository.rb [hunk 1]
```diff
@@ -1,17 +1,42 @@
...
-    ghid:integer:index
-    enabled:boolean
-    secret_key:string
-=end
+    #     ghid:integer:index
+    #     enabled:boolean
+    #     secret_key:string
+
+    def enable_repository
+        client = Octokit::Client.new client_id: ENV['GH_CLIENT_ID'], client_secret: ENV['GH_CLIENT_SECRET']
```
Comment (johnsgill3) [inline app/models/repository.rb]: Just realized I never call `self.update(enabled: true)`. Need to add that call

**Comment by johnsgill3** [comment]
@ryanbmilbourne I think some of the TODO items might have to get turned into bikeshedding issues to be handled when the WebUI is designed unless you have any good ideas at the moment.

**Comment by ryanbmilbourne** [comment]
I think that's fair.  I'll be working on the WebUI this week.

**Comment by ryanbmilbourne** [comment]
@johnsgill3 I see a lot of duplicate commits there.  Total count is at 46 now.  Are you sure you're properly rebasing and squashing when doing your fixups?

**Comment by johnsgill3** [comment]
@ryanbmilbourne Okay I think i squashed it all down into a single commit ready for merge :smile:

**Comment by ryanbmilbourne** [comment]
thanks for squashing!

**Pull request merged**